src/sentry/incidents/models/alert_rule_activations.py

```diff
@@ -63,7 +65,9 @@ class AlertRuleActivations(Model):
     # metric value represents the query results at the end of the activated time window.
     # since activated alerts are not run on a continuously shifting time window, the value
     # of this metric value will only continually tick upwards until the monitor window has expired.
-    metric_value = models.FloatField()
+    metric_value = models.FloatField(null=True)
+    # query_subscriptions are cleaned up after every run, so we keep reference to the ids rather than foreign keying
+    query_subscription_id = models.IntegerField()
```

Comment:
Could you explain why we keep the `query_subscription_id` here after deleting the related subscription? I assume you would prefer to set this null and allow the column to be nullable. In that case, should we keep the FK relation and set the cascade action to SET_NONE?

Reply:
🤔 hmmmmm
I guess i was just thinking there should never be an activation _without_ a query_subscription
so i kind of wanted to ensure that we never find ourselves in a state where that happens? 

I want to avoid a case where we could potentially have a zombie instance and the subscription fails to be created somehow but the activation exists, or something 🤔

Reply:
Chatted w/ Dan over slack

we'll move with the FK relationship and nullable field instead.
This means there'll be a sort of relationship mapping of
```
AlertRule 1-1 SnubaQuery
AlertRule 1-M AlertRuleActivation
AlertRuleActivation - 1-1 QuerySubscription
SnubaQuery 1-M QuerySubscription
```